### PR TITLE
Bugfix: estimates transform correctly when blocks are shifted by a constant

### LINKED ISSUE #43: One more invariance test
Could we please also add the following invariance test to SDID: If we add a constant c to all exposed units with t > T_0 (i.e., to all treated cells), then treatment effects go up by c. This invariance should hold for all methods considered (DID, SC, SDID).

### PR DESCRIPTION
**Pull request by erikcs**
1 and 4 is not as numerically robust as 2 and 3

(closes #43)

### CODE REVIEW

File: tests/testthat/test_synthdid.R [hunk 1]
```diff
@@ -110,3 +110,58 @@ test_that("invariances hold with default options", {
...
+
+  for (c in c(1e-6, 0.25, 1e6)) {
+    # 1.
+    # Re-mapping Yit <- Yit + c for exposed t > T0 increases tau by c
+    Y1 <- setup$Y
+    Y1[exposed, (T0+1):T] <- c + Y1[exposed, (T0+1):T]
+    for (estimator in estimators) {
+      estimate = estimator(Y.orig, N0, T0)
+      estimate.shift = estimator(Y1, N0, T0)
+      expect_equal(c(estimate.shift), c(estimate) + c, tol = 1e-2)
```
Comment (swager) [inline tests/testthat/test_synthdid.R:134]: Ideally we'd hope this goes through with a smaller tolerance. It'd be good to understand why it doesn't.

Comment (erikcs) [reply tests/testthat/test_synthdid.R:134]: Pinging @davidahirshberg here...

Comment (erikcs) [reply tests/testthat/test_synthdid.R:134]: @davidahirshberg is the Franke-Wolfe solver supposed to be translation invariant? Would fitting on standardized data be a sensible approach?

Comment (erikcs) [reply tests/testthat/test_synthdid.R:134]: @davidahirshberg is the Franke-Wolfe solver supposed to be translation invariant? Would fitting on standardized data be a sensible approach?

Comment (davidahirshberg) [reply tests/testthat/test_synthdid.R:134]: Unless there is a bug, this shift should not change the unit or time weights estimated. Neither uses the treated observations in their training.

Comment (davidahirshberg) [reply tests/testthat/test_synthdid.R:134]: The only change from one call to the next should be in this[ line](https://github.com/synth-inference/synthdid/blob/7ff2bd129383833cf8efe89a0cdcc3a6034dfe7c/R/synthdid.R#L68) of code. 
And some entries of Yc that are never looked at.

File: tests/testthat/test_synthdid.R [hunk 2]
```diff
@@ -110,3 +110,58 @@ test_that("invariances hold with default options", {
...
+    Y3 <- setup$Y
+    Y3[!exposed, 1:T0] <- c + Y3[!exposed, 1:T0]
+    for (estimator in estimators[-1]) {
+      estimate = estimator(Y.orig, N0, T0)
+      estimate.shift = estimator(Y3, N0, T0)
+      expect_equal(c(estimate.shift), c(estimate) + c, tol = 1e-10)
+    }
+
+    # 4.
+    # Re-mapping Yit <- Yit + c for unexposed t > T0 decreases tau by c
```
Comment (davidahirshberg) [inline tests/testthat/test_synthdid.R:158]: should be: unexposed t <= T0

File: tests/testthat/test_synthdid.R [hunk 3]
```diff
@@ -110,3 +110,58 @@ test_that("invariances hold with default options", {
...
+    Y4 <- setup$Y
+    Y4[!exposed, (T0+1):T] <- c + Y4[!exposed, (T0+1):T]
+    for (estimator in estimators) {
+      estimate = estimator(Y.orig, N0, T0)
+      estimate.shift = estimator(Y4, N0, T0)
+      expect_equal(c(estimate.shift), c(estimate) - c, tol = 1e-2)
+    }
+  }
+})
```
Comment (davidahirshberg) [inline tests/testthat/test_synthdid.R:167]: Synthetic control isn't invariant to this kind of transformation.  The other two should be.

Comment (erikcs) [reply tests/testthat/test_synthdid.R:167]: The branch is `inv2` in the main repo - feel free to check it out and push to it

File: tests/testthat/test_synthdid.R (hunk 1 above)
Comment (davidahirshberg) [reply tests/testthat/test_synthdid.R]: There was a bug. This translation was affecting the default value of the tuning parameter zeta.omega in sc_estimate. We'd fixed the same bug earlier for synthdid_estimate.

**Review by davidahirshberg — APPROVED** [review]
Looks good.

**Pull request merged**